submissions/krylenger/memory-pair-game/app.js

```diff
@@ -0,0 +1,244 @@
+    },
+    {
+        key: 5,
+        text: 'dim',
+        musicCard: false
+    },
+    {
+        key: 5,
+        text: '$',
+        musicCard: true,
+        song: dimSong
+    }
+]
+
+const main = document.querySelector('.main');
+let mainGridContainer = document.querySelector('.main__grid-container');
+const modal = document.querySelector('.modal');
+
+
+const styleNewCardFlipperFront = (newCardFlipperFront, card, cardNumber) => {
+    switch (cardNumber) {
+        case 0:
+        case 9:
+            newCardFlipperFront.classList.add('card__flipper-front', card.key, `card--circle-img`);
+            break;
+        case 1:
+        case 2:
+        case 7:
+        case 8:
+            newCardFlipperFront.classList.add('card__flipper-front', card.key, `card--vertical-line-img`);
+            break;
+        case 3:
+            newCardFlipperFront.classList.add('card__flipper-front', card.key, `card--left-corner-img`);
+            break;
+        case 6:
+            newCardFlipperFront.classList.add('card__flipper-front', card.key, `card--right-corner-img`);
+            break;
+        case 4:
+        case 5:
+            newCardFlipperFront.classList.add('card__flipper-front', card.key, `card--horizontal-line-img`);
+            break;
+        default:
+            newCardFlipperFront.classList.add('card__flipper-front', card.key, `card--clef-img`);
+    }
+    
+}
+
+const createNewCard = (card, cardNumber) => {
+    console.log(cardNumber);
+    const newCard = document.createElement('div');
+    const newCardFlipper = document.createElement('div');
+    const newCardFlipperFront = document.createElement('div');
+    const newCardFlipperBack = document.createElement('div');
+    if (card.musicCard) {
+        newCard.classList.add('card', `card--${cardNumber}`, 'musicCard')
+        newCardFlipperBack.classList.add(`card--music-img`);
+    } else {
+        newCard.classList.add('card', `card--${cardNumber}`);
+        newCardFlipperBack.classList.add(`card--${card.text}-img`);
+    }
+    newCardFlipper.classList.add('card__flipper');
+    newCardFlipper.setAttribute('ontouchstart', "this.classList.toggle('card__flipper--flip');")
+    styleNewCardFlipperFront(newCardFlipperFront, card, cardNumber);
+    newCardFlipperBack.classList.add('card__flipper-back');
+    newCard.appendChild(newCardFlipper);
+    newCardFlipper.appendChild(newCardFlipperFront);
+    newCardFlipper.appendChild(newCardFlipperBack);
+    return newCard;
+}
+
+const shuffleArr = (arr) => {
+    return arr.sort(() => 0.5 - Math.random());
+}
+
+const loadCards = (cardsArr) => {
+    const shuffledCards = shuffleArr(cardsArr);
+    const fragment = document.createDocumentFragment();
+    shuffledCards.forEach((card, index) => {
+        const newCard = createNewCard(card, index);
+        fragment.appendChild(newCard);
+    })
+    mainGridContainer.appendChild(fragment);
+}
+
+const recreateMainGridContainer = () => {
+    mainGridContainer.remove();
```

Comment:
You can use `mainGridContainer.innerHTML = ''`
It'll help you not to remove this element from the DOM and create it again

Reply:
Thanks! This helped me to remove several lines of code + one function.
✅ Done

Reply:
Yep. It happened because you called `flipCards` function which added a new `click` event listener to the  `mainGridContainer` DOM element. You can check how many event listeners each DOM element has by expanding the `event` sign in Firefox (maybe in Chrome you can find it as well)
<img width="1228" alt="Screenshot 2020-11-23 at 14 11 11" src="https://user-images.githubusercontent.com/13544983/99961481-32c4c500-2d97-11eb-8ac9-a0781fc8deb1.png">
So on each click, your script called two event listeners.
Also, it seems that the `playingCards` variable isn't reset. Be careful with the variables and pay attention to their scope in the `handleGameOver` function. I think the problem is there.
Hope it will help you to fix this

Reply:
Thank you one more time for your help!
It really helps me to become better every time!
After your feedback, I've redesigned my code a little bit (please have a look) by moving callback of eventListener into separate function.
This helped me to remove the callback from the 1st round when I'm resetting game to play the 2nd round. 
And now I clearly see that the design of my app is more consistent and logical. 
Appreciate!

Reply:
Cool! I'm glad that it helped you! 👍🏻

Reply:
so are we done with this project?

Reply:
I don't like the approach of adding the `card.key` to the class names. I'd recommend using data attributes for this.
If it doesn't take much time for you, cosider refactoring functions which depend on it, please

Reply:
Again, new stuff to learn, thanks!)
✅ Done